src/framework/dockwindow/internal/dockbase.cpp

```diff
@@ -449,7 +454,13 @@ void DockBase::open()
         return;
     }
 
-    m_dockWidget->show();
+    // If the widget is in the frame (was not closed), just ask the frame to make the widget's tab active.
+    // Otherwise show the widget. This will add it back into the frame or float it. It will also make it active.
+    if (m_dockWidget->frame() && m_dockWidget->frame()->containsDockWidget(m_dockWidget)) {
+        m_dockWidget->frame()->setCurrentDockWidget(m_dockWidget);
```

Comment:
Does this perhaps make the usage of `makeCurrentTabInFrame` in `DockPageView::setDockOpen` redundant? (seems preferable to have that kind of logic not in multiple places when one place is enough)

Reply:
Hm, never mind, in `setDockOpen` it does something else than what I was expecting. Perhaps I was confused because I had seen this kind of logic in that method in https://github.com/musescore/MuseScore/pull/15675 or yet another unmerged PR.